### PR TITLE
Exclude the `object` possibility from the `ClientSession.timeout` type

### PR DESCRIPTION
**Pull request by DevilXD**
## What do these changes do?

This is a follow-up PR to #6917, which fixes the overly inclusive typing of `ClientSession.timeout` attribute, which can never actually end up as an `object` instance. This leftover probably exists from the time when there was the `_sentinel` value being assigned to the attribute (see "3.8 branch `self._timeout` assignment" from the linked issue), hence why it ended up this way.

## Are there changes in behavior for the user?

For untyped code, there's absolutely no change - the `self._timeout` attribute was never actually assigned the sentinel value to. Most of the explanation already exists in #6917, here's the test code snippet from there again:

```py
import asyncio
import aiohttp

async def main:
    session = aiohttp.ClientSession(timeout=aiohttp.ClientTimeout(total=10))
    # read back the total time attribute
    total_time = session.timeout.total  # "object" type of "Union[object, ClientTimeout]" has no attribute "total"
    print(total_time)

asyncio.run(main())
```

For typed code, the current solution would be to `assert isinstance(session.timeout, aiohttp.ClientTimeout)` everywhere the attribute is being accessed. This PR removes this "unnecessary necessity".

## Related issue number

#6917

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [x] Documentation reflects the changes (documentation never mentioned an `object` instance being accessible from there even)
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by DevilXD** [comment]
I'm not sure if I deserve a `CONTRIBUTORS` entry for such a simple typing-only change (it specifically asks for "code modification" and this changes nothing other than typing), so I left it out for now (and take no issue with that). If it would be needed though, please let me know and I'll add a commit for it.

File: CHANGES/6917.misc [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix ClientSession.timeout having incorrect typing.
```
Comment (webknjaz) [inline CHANGES/6917.misc]: Plz try following the recommendations at https://github.com/aio-libs/aiohttp/tree/master/CHANGES#alright-so-how-to-add-a-news-fragment. Also, use RST formatting to highlight/link identifiers.

**Comment by Dreamsorcerer** [comment]
> I'm not sure if I deserve a `CONTRIBUTORS` entry...

I've never even added my own name. :P
Not sure that file is really useful, personally

**Comment by webknjaz** [comment]
> Merge branch 'master' into patch-1

@Dreamsorcerer FYI you can now select a rebase mode in that GH button (just click on the drop-down arrow)

**Comment by Dreamsorcerer** [comment]
> @Dreamsorcerer FYI you can now select a rebase mode in that GH button (just click on the drop-down arrow)

Ah, it all get squash committed anyway though, so I'm not it makes any difference.

**Review by Dreamsorcerer — APPROVED** [review]
This seems obviously correct to me, probably didn't need the long story in the bug ticket. :P

**Comment by Dreamsorcerer** [comment]
@webknjaz Seems like a simple fix for 3.8.2 if we've not already passed the point of no return?

**Comment by webknjaz** [comment]
> This seems obviously correct to me, probably didn't need the long story in the bug ticket. :P

A good description is useful, since what's obvious for one person would be confusing for another. Especially, this applies to a “future you”, or a “trying to decide on backports you”, or a “Git archeologist-you”.

File: CHANGES/6917.misc (hunk 1 above)
Comment (webknjaz) [reply CHANGES/6917.misc]: Also, this is probably a bugfix + maybe docs type.

**Comment by webknjaz** [comment]
> @webknjaz Seems like a simple fix for 3.8.2 if we've not already passed the point of no return?

I guess I could get it in. But it still needs brushing up the change note.

File: CHANGES/6917.misc (hunk 1 above)
Comment (webknjaz) [reply CHANGES/6917.misc]: I've made the updates myself.

**Pull request merged**